<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">

<head>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1"/>
  <meta name="generator" content="distill" />

  <style type="text/css">
  /* Hide doc at startup (prevent jankiness while JS renders/transforms) */
  body {
    visibility: hidden;
  }
  </style>

 <!--radix_placeholder_import_source-->
 <!--/radix_placeholder_import_source-->

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  { color: #00769e; background-color: #f1f3f5; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #00769e; } /* Normal */
code span.al { color: #ad0000; } /* Alert */
code span.an { color: #5e5e5e; } /* Annotation */
code span.at { color: #657422; } /* Attribute */
code span.bn { color: #ad0000; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #00769e; } /* ControlFlow */
code span.ch { color: #20794d; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #5e5e5e; } /* Comment */
code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
code span.dt { color: #ad0000; } /* DataType */
code span.dv { color: #ad0000; } /* DecVal */
code span.er { color: #ad0000; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #ad0000; } /* Float */
code span.fu { color: #4758ab; } /* Function */
code span.im { } /* Import */
code span.in { color: #5e5e5e; } /* Information */
code span.kw { color: #00769e; } /* Keyword */
code span.op { color: #5e5e5e; } /* Operator */
code span.ot { color: #00769e; } /* Other */
code span.pp { color: #ad0000; } /* Preprocessor */
code span.sc { color: #5e5e5e; } /* SpecialChar */
code span.ss { color: #20794d; } /* SpecialString */
code span.st { color: #20794d; } /* String */
code span.va { color: #111111; } /* Variable */
code span.vs { color: #20794d; } /* VerbatimString */
code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
</style>


  <!--radix_placeholder_meta_tags-->
<title>Tim Barry: Genome-wide association studies</title>

<meta property="description" itemprop="description" content="An overview of genome-wide association studies."/>


<!--  https://schema.org/Article -->
<meta property="article:published" itemprop="datePublished" content="2024-05-01"/>
<meta property="article:created" itemprop="dateCreated" content="2024-05-01"/>
<meta name="article:author" content="Tim Barry"/>

<!--  https://developers.facebook.com/docs/sharing/webmasters#markup -->
<meta property="og:title" content="Tim Barry: Genome-wide association studies"/>
<meta property="og:type" content="article"/>
<meta property="og:description" content="An overview of genome-wide association studies."/>
<meta property="og:locale" content="en_US"/>
<meta property="og:site_name" content="Tim Barry"/>

<!--  https://dev.twitter.com/cards/types/summary -->
<meta property="twitter:card" content="summary"/>
<meta property="twitter:title" content="Tim Barry: Genome-wide association studies"/>
<meta property="twitter:description" content="An overview of genome-wide association studies."/>

<!--/radix_placeholder_meta_tags-->
  <!--radix_placeholder_rmarkdown_metadata-->

<script type="text/json" id="radix-rmarkdown-metadata">
{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["title","description","author","date","output","categories","draft","preview"]}},"value":[{"type":"character","attributes":{},"value":["Genome-wide association studies"]},{"type":"character","attributes":{},"value":["An overview of genome-wide association studies."]},{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","url"]}},"value":[{"type":"character","attributes":{},"value":["Tim Barry"]},{"type":"character","attributes":{},"value":["https://timothy-barry.github.io"]}]}]},{"type":"character","attributes":{},"value":["05-01-2024"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["distill::distill_article"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["self_contained"]}},"value":[{"type":"logical","attributes":{},"value":[false]}]}]},{"type":"character","attributes":{},"value":["Genomics"]},{"type":"logical","attributes":{},"value":[false]},{"type":"character","attributes":{},"value":["size_vs_maf.png"]}]}
</script>
<!--/radix_placeholder_rmarkdown_metadata-->
  
  <script type="text/json" id="radix-resource-manifest">
  {"type":"character","attributes":{},"value":["genome_1.png","gwas_files/anchor-4.2.2/anchor.min.js","gwas_files/bowser-1.9.3/bowser.min.js","gwas_files/distill-2.2.21/template.v2.js","gwas_files/header-attrs-2.26/header-attrs.js","gwas_files/jquery-3.6.0/jquery-3.6.0.js","gwas_files/jquery-3.6.0/jquery-3.6.0.min.js","gwas_files/jquery-3.6.0/jquery-3.6.0.min.map","gwas_files/popper-2.6.0/popper.min.js","gwas_files/tippy-6.2.7/tippy-bundle.umd.min.js","gwas_files/tippy-6.2.7/tippy-light-border.css","gwas_files/tippy-6.2.7/tippy.css","gwas_files/tippy-6.2.7/tippy.umd.min.js","gwas_files/webcomponents-2.0.0/webcomponents.js","sex_repro.png","size_vs_maf.png","snp_matrix_2.png","snp_matrix.png"]}
  </script>
  <!--radix_placeholder_navigation_in_header-->
<meta name="distill:offset" content="../.."/>

<script type="application/javascript">

  window.headroom_prevent_pin = false;

  window.document.addEventListener("DOMContentLoaded", function (event) {

    // initialize headroom for banner
    var header = $('header').get(0);
    var headerHeight = header.offsetHeight;
    var headroom = new Headroom(header, {
      tolerance: 5,
      onPin : function() {
        if (window.headroom_prevent_pin) {
          window.headroom_prevent_pin = false;
          headroom.unpin();
        }
      }
    });
    headroom.init();
    if(window.location.hash)
      headroom.unpin();
    $(header).addClass('headroom--transition');

    // offset scroll location for banner on hash change
    // (see: https://github.com/WickyNilliams/headroom.js/issues/38)
    window.addEventListener("hashchange", function(event) {
      window.scrollTo(0, window.pageYOffset - (headerHeight + 25));
    });

    // responsive menu
    $('.distill-site-header').each(function(i, val) {
      var topnav = $(this);
      var toggle = topnav.find('.nav-toggle');
      toggle.on('click', function() {
        topnav.toggleClass('responsive');
      });
    });

    // nav dropdowns
    $('.nav-dropbtn').click(function(e) {
      $(this).next('.nav-dropdown-content').toggleClass('nav-dropdown-active');
      $(this).parent().siblings('.nav-dropdown')
         .children('.nav-dropdown-content').removeClass('nav-dropdown-active');
    });
    $("body").click(function(e){
      $('.nav-dropdown-content').removeClass('nav-dropdown-active');
    });
    $(".nav-dropdown").click(function(e){
      e.stopPropagation();
    });
  });
</script>

<style type="text/css">

/* Theme (user-documented overrideables for nav appearance) */

.distill-site-nav {
  color: rgba(255, 255, 255, 0.8);
  background-color: #0F2E3D;
  font-size: 15px;
  font-weight: 300;
}

.distill-site-nav a {
  color: inherit;
  text-decoration: none;
}

.distill-site-nav a:hover {
  color: white;
}

@media print {
  .distill-site-nav {
    display: none;
  }
}

.distill-site-header {

}

.distill-site-footer {

}


/* Site Header */

.distill-site-header {
  width: 100%;
  box-sizing: border-box;
  z-index: 3;
}

.distill-site-header .nav-left {
  display: inline-block;
  margin-left: 8px;
}

@media screen and (max-width: 768px) {
  .distill-site-header .nav-left {
    margin-left: 0;
  }
}


.distill-site-header .nav-right {
  float: right;
  margin-right: 8px;
}

.distill-site-header a,
.distill-site-header .title {
  display: inline-block;
  text-align: center;
  padding: 14px 10px 14px 10px;
}

.distill-site-header .title {
  font-size: 18px;
  min-width: 150px;
}

.distill-site-header .logo {
  padding: 0;
}

.distill-site-header .logo img {
  display: none;
  max-height: 20px;
  width: auto;
  margin-bottom: -4px;
}

.distill-site-header .nav-image img {
  max-height: 18px;
  width: auto;
  display: inline-block;
  margin-bottom: -3px;
}



@media screen and (min-width: 1000px) {
  .distill-site-header .logo img {
    display: inline-block;
  }
  .distill-site-header .nav-left {
    margin-left: 20px;
  }
  .distill-site-header .nav-right {
    margin-right: 20px;
  }
  .distill-site-header .title {
    padding-left: 12px;
  }
}


.distill-site-header .nav-toggle {
  display: none;
}

.nav-dropdown {
  display: inline-block;
  position: relative;
}

.nav-dropdown .nav-dropbtn {
  border: none;
  outline: none;
  color: rgba(255, 255, 255, 0.8);
  padding: 16px 10px;
  background-color: transparent;
  font-family: inherit;
  font-size: inherit;
  font-weight: inherit;
  margin: 0;
  margin-top: 1px;
  z-index: 2;
}

.nav-dropdown-content {
  display: none;
  position: absolute;
  background-color: white;
  min-width: 200px;
  border: 1px solid rgba(0,0,0,0.15);
  border-radius: 4px;
  box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.1);
  z-index: 1;
  margin-top: 2px;
  white-space: nowrap;
  padding-top: 4px;
  padding-bottom: 4px;
}

.nav-dropdown-content hr {
  margin-top: 4px;
  margin-bottom: 4px;
  border: none;
  border-bottom: 1px solid rgba(0, 0, 0, 0.1);
}

.nav-dropdown-active {
  display: block;
}

.nav-dropdown-content a, .nav-dropdown-content .nav-dropdown-header {
  color: black;
  padding: 6px 24px;
  text-decoration: none;
  display: block;
  text-align: left;
}

.nav-dropdown-content .nav-dropdown-header {
  display: block;
  padding: 5px 24px;
  padding-bottom: 0;
  text-transform: uppercase;
  font-size: 14px;
  color: #999999;
  white-space: nowrap;
}

.nav-dropdown:hover .nav-dropbtn {
  color: white;
}

.nav-dropdown-content a:hover {
  background-color: #ddd;
  color: black;
}

.nav-right .nav-dropdown-content {
  margin-left: -45%;
  right: 0;
}

@media screen and (max-width: 768px) {
  .distill-site-header a, .distill-site-header .nav-dropdown  {display: none;}
  .distill-site-header a.nav-toggle {
    float: right;
    display: block;
  }
  .distill-site-header .title {
    margin-left: 0;
  }
  .distill-site-header .nav-right {
    margin-right: 0;
  }
  .distill-site-header {
    overflow: hidden;
  }
  .nav-right .nav-dropdown-content {
    margin-left: 0;
  }
}


@media screen and (max-width: 768px) {
  .distill-site-header.responsive {position: relative; min-height: 500px; }
  .distill-site-header.responsive a.nav-toggle {
    position: absolute;
    right: 0;
    top: 0;
  }
  .distill-site-header.responsive a,
  .distill-site-header.responsive .nav-dropdown {
    display: block;
    text-align: left;
  }
  .distill-site-header.responsive .nav-left,
  .distill-site-header.responsive .nav-right {
    width: 100%;
  }
  .distill-site-header.responsive .nav-dropdown {float: none;}
  .distill-site-header.responsive .nav-dropdown-content {position: relative;}
  .distill-site-header.responsive .nav-dropdown .nav-dropbtn {
    display: block;
    width: 100%;
    text-align: left;
  }
}

/* Site Footer */

.distill-site-footer {
  width: 100%;
  overflow: hidden;
  box-sizing: border-box;
  z-index: 3;
  margin-top: 30px;
  padding-top: 30px;
  padding-bottom: 30px;
  text-align: center;
}

/* Headroom */

d-title {
  padding-top: 6rem;
}

@media print {
  d-title {
    padding-top: 4rem;
  }
}

.headroom {
  z-index: 1000;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
}

.headroom--transition {
  transition: all .4s ease-in-out;
}

.headroom--unpinned {
  top: -100px;
}

.headroom--pinned {
  top: 0;
}

/* adjust viewport for navbar height */
/* helps vertically center bootstrap (non-distill) content */
.min-vh-100 {
  min-height: calc(100vh - 100px) !important;
}

</style>

<script src="../../site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="../../site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"/>
<link href="../../site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"/>
<script src="../../site_libs/headroom-0.9.4/headroom.min.js"></script>
<!--/radix_placeholder_navigation_in_header-->
  <!--radix_placeholder_distill-->

<style type="text/css">

body {
  background-color: white;
}

.pandoc-table {
  width: 100%;
}

.pandoc-table>caption {
  margin-bottom: 10px;
}

.pandoc-table th:not([align]) {
  text-align: left;
}

.pagedtable-footer {
  font-size: 15px;
}

d-byline .byline {
  grid-template-columns: 2fr 2fr;
}

d-byline .byline h3 {
  margin-block-start: 1.5em;
}

d-byline .byline .authors-affiliations h3 {
  margin-block-start: 0.5em;
}

.authors-affiliations .orcid-id {
  width: 16px;
  height:16px;
  margin-left: 4px;
  margin-right: 4px;
  vertical-align: middle;
  padding-bottom: 2px;
}

d-title .dt-tags {
  margin-top: 1em;
  grid-column: text;
}

.dt-tags .dt-tag {
  text-decoration: none;
  display: inline-block;
  color: rgba(0,0,0,0.6);
  padding: 0em 0.4em;
  margin-right: 0.5em;
  margin-bottom: 0.4em;
  font-size: 70%;
  border: 1px solid rgba(0,0,0,0.2);
  border-radius: 3px;
  text-transform: uppercase;
  font-weight: 500;
}

d-article table.gt_table td,
d-article table.gt_table th {
  border-bottom: none;
  font-size: 100%;
}

.html-widget {
  margin-bottom: 2.0em;
}

.l-screen-inset {
  padding-right: 16px;
}

.l-screen .caption {
  margin-left: 10px;
}

.shaded {
  background: rgb(247, 247, 247);
  padding-top: 20px;
  padding-bottom: 20px;
  border-top: 1px solid rgba(0, 0, 0, 0.1);
  border-bottom: 1px solid rgba(0, 0, 0, 0.1);
}

.shaded .html-widget {
  margin-bottom: 0;
  border: 1px solid rgba(0, 0, 0, 0.1);
}

.shaded .shaded-content {
  background: white;
}

.text-output {
  margin-top: 0;
  line-height: 1.5em;
}

.hidden {
  display: none !important;
}

hr.section-separator {
  border: none;
  border-top: 1px solid rgba(0, 0, 0, 0.1);
  margin: 0px;
}


d-byline {
  border-top: none;
}

d-article {
  padding-top: 2.5rem;
  padding-bottom: 30px;
  border-top: none;
}

d-appendix {
  padding-top: 30px;
}

d-article>p>img {
  width: 100%;
}

d-article h2 {
  margin: 1rem 0 1.5rem 0;
}

d-article h3 {
  margin-top: 1.5rem;
}

d-article iframe {
  border: 1px solid rgba(0, 0, 0, 0.1);
  margin-bottom: 2.0em;
  width: 100%;
}

/* Tweak code blocks */

d-article div.sourceCode code,
d-article pre code {
  font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
}

d-article pre,
d-article div.sourceCode,
d-article div.sourceCode pre {
  overflow: auto;
}

d-article div.sourceCode {
  background-color: white;
}

d-article div.sourceCode pre {
  padding-left: 10px;
  font-size: 12px;
  border-left: 2px solid rgba(0,0,0,0.1);
}

d-article pre {
  font-size: 12px;
  color: black;
  background: none;
  margin-top: 0;
  text-align: left;
  white-space: pre;
  word-spacing: normal;
  word-break: normal;
  word-wrap: normal;
  line-height: 1.5;

  -moz-tab-size: 4;
  -o-tab-size: 4;
  tab-size: 4;

  -webkit-hyphens: none;
  -moz-hyphens: none;
  -ms-hyphens: none;
  hyphens: none;
}

d-article pre a {
  border-bottom: none;
}

d-article pre a:hover {
  border-bottom: none;
  text-decoration: underline;
}

d-article details {
  grid-column: text;
  margin-bottom: 0.8em;
}

@media(min-width: 768px) {

d-article pre,
d-article div.sourceCode,
d-article div.sourceCode pre {
  overflow: visible !important;
}

d-article div.sourceCode pre {
  padding-left: 18px;
  font-size: 14px;
}

/* tweak for Pandoc numbered line within distill */
d-article pre.numberSource code > span {
    left: -2em;
}

d-article pre {
  font-size: 14px;
}

}

figure img.external {
  background: white;
  border: 1px solid rgba(0, 0, 0, 0.1);
  box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
  padding: 18px;
  box-sizing: border-box;
}

/* CSS for d-contents */

.d-contents {
  grid-column: text;
  color: rgba(0,0,0,0.8);
  font-size: 0.9em;
  padding-bottom: 1em;
  margin-bottom: 1em;
  padding-bottom: 0.5em;
  margin-bottom: 1em;
  padding-left: 0.25em;
  justify-self: start;
}

@media(min-width: 1000px) {
  .d-contents.d-contents-float {
    height: 0;
    grid-column-start: 1;
    grid-column-end: 4;
    justify-self: center;
    padding-right: 3em;
    padding-left: 2em;
  }
}

.d-contents nav h3 {
  font-size: 18px;
  margin-top: 0;
  margin-bottom: 1em;
}

.d-contents li {
  list-style-type: none
}

.d-contents nav > ul {
  padding-left: 0;
}

.d-contents ul {
  padding-left: 1em
}

.d-contents nav ul li {
  margin-top: 0.6em;
  margin-bottom: 0.2em;
}

.d-contents nav a {
  font-size: 13px;
  border-bottom: none;
  text-decoration: none
  color: rgba(0, 0, 0, 0.8);
}

.d-contents nav a:hover {
  text-decoration: underline solid rgba(0, 0, 0, 0.6)
}

.d-contents nav > ul > li > a {
  font-weight: 600;
}

.d-contents nav > ul > li > ul {
  font-weight: inherit;
}

.d-contents nav > ul > li > ul > li {
  margin-top: 0.2em;
}


.d-contents nav ul {
  margin-top: 0;
  margin-bottom: 0.25em;
}

.d-article-with-toc h2:nth-child(2) {
  margin-top: 0;
}


/* Figure */

.figure {
  position: relative;
  margin-bottom: 2.5em;
  margin-top: 1.5em;
}

.figure .caption {
  color: rgba(0, 0, 0, 0.6);
  font-size: 12px;
  line-height: 1.5em;
}

.figure img.external {
  background: white;
  border: 1px solid rgba(0, 0, 0, 0.1);
  box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
  padding: 18px;
  box-sizing: border-box;
}

.figure .caption a {
  color: rgba(0, 0, 0, 0.6);
}

.figure .caption b,
.figure .caption strong, {
  font-weight: 600;
  color: rgba(0, 0, 0, 1.0);
}

/* Citations */

d-article .citation {
  color: inherit;
  cursor: inherit;
}

div.hanging-indent{
  margin-left: 1em; text-indent: -1em;
}

/* Citation hover box */

.tippy-box[data-theme~=light-border] {
  background-color: rgba(250, 250, 250, 0.95);
}

.tippy-content > p {
  margin-bottom: 0;
  padding: 2px;
}


/* Tweak 1000px media break to show more text */

@media(min-width: 1000px) {
  .base-grid,
  distill-header,
  d-title,
  d-abstract,
  d-article,
  d-appendix,
  distill-appendix,
  d-byline,
  d-footnote-list,
  d-citation-list,
  distill-footer {
    grid-template-columns: [screen-start] 1fr [page-start kicker-start] 80px [middle-start] 50px [text-start kicker-end] 65px 65px 65px 65px 65px 65px 65px 65px [text-end gutter-start] 65px [middle-end] 65px [page-end gutter-end] 1fr [screen-end];
    grid-column-gap: 16px;
  }

  .grid {
    grid-column-gap: 16px;
  }

  d-article {
    font-size: 1.06rem;
    line-height: 1.7em;
  }
  figure .caption, .figure .caption, figure figcaption {
    font-size: 13px;
  }
}

@media(min-width: 1180px) {
  .base-grid,
  distill-header,
  d-title,
  d-abstract,
  d-article,
  d-appendix,
  distill-appendix,
  d-byline,
  d-footnote-list,
  d-citation-list,
  distill-footer {
    grid-template-columns: [screen-start] 1fr [page-start kicker-start] 60px [middle-start] 60px [text-start kicker-end] 60px 60px 60px 60px 60px 60px 60px 60px [text-end gutter-start] 60px [middle-end] 60px [page-end gutter-end] 1fr [screen-end];
    grid-column-gap: 32px;
  }

  .grid {
    grid-column-gap: 32px;
  }
}


/* Get the citation styles for the appendix (not auto-injected on render since
   we do our own rendering of the citation appendix) */

d-appendix .citation-appendix,
.d-appendix .citation-appendix {
  font-size: 11px;
  line-height: 15px;
  border-left: 1px solid rgba(0, 0, 0, 0.1);
  padding-left: 18px;
  border: 1px solid rgba(0,0,0,0.1);
  background: rgba(0, 0, 0, 0.02);
  padding: 10px 18px;
  border-radius: 3px;
  color: rgba(150, 150, 150, 1);
  overflow: hidden;
  margin-top: -12px;
  white-space: pre-wrap;
  word-wrap: break-word;
}

/* Include appendix styles here so they can be overridden */

d-appendix {
  contain: layout style;
  font-size: 0.8em;
  line-height: 1.7em;
  margin-top: 60px;
  margin-bottom: 0;
  border-top: 1px solid rgba(0, 0, 0, 0.1);
  color: rgba(0,0,0,0.5);
  padding-top: 60px;
  padding-bottom: 48px;
}

d-appendix h3 {
  grid-column: page-start / text-start;
  font-size: 15px;
  font-weight: 500;
  margin-top: 1em;
  margin-bottom: 0;
  color: rgba(0,0,0,0.65);
}

d-appendix h3 + * {
  margin-top: 1em;
}

d-appendix ol {
  padding: 0 0 0 15px;
}

@media (min-width: 768px) {
  d-appendix ol {
    padding: 0 0 0 30px;
    margin-left: -30px;
  }
}

d-appendix li {
  margin-bottom: 1em;
}

d-appendix a {
  color: rgba(0, 0, 0, 0.6);
}

d-appendix > * {
  grid-column: text;
}

d-appendix > d-footnote-list,
d-appendix > d-citation-list,
d-appendix > distill-appendix {
  grid-column: screen;
}

/* Include footnote styles here so they can be overridden */

d-footnote-list {
  contain: layout style;
}

d-footnote-list > * {
  grid-column: text;
}

d-footnote-list a.footnote-backlink {
  color: rgba(0,0,0,0.3);
  padding-left: 0.5em;
}



/* Anchor.js */

.anchorjs-link {
  /*transition: all .25s linear; */
  text-decoration: none;
  border-bottom: none;
}
*:hover > .anchorjs-link {
  margin-left: -1.125em !important;
  text-decoration: none;
  border-bottom: none;
}

/* Social footer */

.social_footer {
  margin-top: 30px;
  margin-bottom: 0;
  color: rgba(0,0,0,0.67);
}

.disqus-comments {
  margin-right: 30px;
}

.disqus-comment-count {
  border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  cursor: pointer;
}

#disqus_thread {
  margin-top: 30px;
}

.article-sharing a {
  border-bottom: none;
  margin-right: 8px;
}

.article-sharing a:hover {
  border-bottom: none;
}

.sidebar-section.subscribe {
  font-size: 12px;
  line-height: 1.6em;
}

.subscribe p {
  margin-bottom: 0.5em;
}


.article-footer .subscribe {
  font-size: 15px;
  margin-top: 45px;
}


.sidebar-section.custom {
  font-size: 12px;
  line-height: 1.6em;
}

.custom p {
  margin-bottom: 0.5em;
}

/* Styles for listing layout (hide title) */
.layout-listing d-title, .layout-listing .d-title {
  display: none;
}

/* Styles for posts lists (not auto-injected) */


.posts-with-sidebar {
  padding-left: 45px;
  padding-right: 45px;
}

.posts-list .description h2,
.posts-list .description p {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
}

.posts-list .description h2 {
  font-weight: 700;
  border-bottom: none;
  padding-bottom: 0;
}

.posts-list h2.post-tag {
  border-bottom: 1px solid rgba(0, 0, 0, 0.2);
  padding-bottom: 12px;
}
.posts-list {
  margin-top: 60px;
  margin-bottom: 24px;
}

.posts-list .post-preview {
  text-decoration: none;
  overflow: hidden;
  display: block;
  border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  padding: 24px 0;
}

.post-preview-last {
  border-bottom: none !important;
}

.posts-list .posts-list-caption {
  grid-column: screen;
  font-weight: 400;
}

.posts-list .post-preview h2 {
  margin: 0 0 6px 0;
  line-height: 1.2em;
  font-style: normal;
  font-size: 24px;
}

.posts-list .post-preview p {
  margin: 0 0 12px 0;
  line-height: 1.4em;
  font-size: 16px;
}

.posts-list .post-preview .thumbnail {
  box-sizing: border-box;
  margin-bottom: 24px;
  position: relative;
  max-width: 500px;
}
.posts-list .post-preview img {
  width: 100%;
  display: block;
}

.posts-list .metadata {
  font-size: 12px;
  line-height: 1.4em;
  margin-bottom: 18px;
}

.posts-list .metadata > * {
  display: inline-block;
}

.posts-list .metadata .publishedDate {
  margin-right: 2em;
}

.posts-list .metadata .dt-authors {
  display: block;
  margin-top: 0.3em;
  margin-right: 2em;
}

.posts-list .dt-tags {
  display: block;
  line-height: 1em;
}

.posts-list .dt-tags .dt-tag {
  display: inline-block;
  color: rgba(0,0,0,0.6);
  padding: 0.3em 0.4em;
  margin-right: 0.2em;
  margin-bottom: 0.4em;
  font-size: 60%;
  border: 1px solid rgba(0,0,0,0.2);
  border-radius: 3px;
  text-transform: uppercase;
  font-weight: 500;
}

.posts-list img {
  opacity: 1;
}

.posts-list img[data-src] {
  opacity: 0;
}

.posts-more {
  clear: both;
}


.posts-sidebar {
  font-size: 16px;
}

.posts-sidebar h3 {
  font-size: 16px;
  margin-top: 0;
  margin-bottom: 0.5em;
  font-weight: 400;
  text-transform: uppercase;
}

.sidebar-section {
  margin-bottom: 30px;
}

.categories ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

.categories li {
  color: rgba(0, 0, 0, 0.8);
  margin-bottom: 0;
}

.categories li>a {
  border-bottom: none;
}

.categories li>a:hover {
  border-bottom: 1px solid rgba(0, 0, 0, 0.4);
}

.categories .active {
  font-weight: 600;
}

.categories .category-count {
  color: rgba(0, 0, 0, 0.4);
}


@media(min-width: 768px) {
  .posts-list .post-preview h2 {
    font-size: 26px;
  }
  .posts-list .post-preview .thumbnail {
    float: right;
    width: 30%;
    margin-bottom: 0;
  }
  .posts-list .post-preview .description {
    float: left;
    width: 45%;
  }
  .posts-list .post-preview .metadata {
    float: left;
    width: 20%;
    margin-top: 8px;
  }
  .posts-list .post-preview p {
    margin: 0 0 12px 0;
    line-height: 1.5em;
    font-size: 16px;
  }
  .posts-with-sidebar .posts-list {
    float: left;
    width: 75%;
  }
  .posts-with-sidebar .posts-sidebar {
    float: right;
    width: 20%;
    margin-top: 60px;
    padding-top: 24px;
    padding-bottom: 24px;
  }
}


/* Improve display for browsers without grid (IE/Edge <= 15) */

.downlevel {
  line-height: 1.6em;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", Arial, sans-serif;
  margin: 0;
}

.downlevel .d-title {
  padding-top: 6rem;
  padding-bottom: 1.5rem;
}

.downlevel .d-title h1 {
  font-size: 50px;
  font-weight: 700;
  line-height: 1.1em;
  margin: 0 0 0.5rem;
}

.downlevel .d-title p {
  font-weight: 300;
  font-size: 1.2rem;
  line-height: 1.55em;
  margin-top: 0;
}

.downlevel .d-byline {
  padding-top: 0.8em;
  padding-bottom: 0.8em;
  font-size: 0.8rem;
  line-height: 1.8em;
}

.downlevel .section-separator {
  border: none;
  border-top: 1px solid rgba(0, 0, 0, 0.1);
}

.downlevel .d-article {
  font-size: 1.06rem;
  line-height: 1.7em;
  padding-top: 1rem;
  padding-bottom: 2rem;
}


.downlevel .d-appendix {
  padding-left: 0;
  padding-right: 0;
  max-width: none;
  font-size: 0.8em;
  line-height: 1.7em;
  margin-bottom: 0;
  color: rgba(0,0,0,0.5);
  padding-top: 40px;
  padding-bottom: 48px;
}

.downlevel .footnotes ol {
  padding-left: 13px;
}

.downlevel .base-grid,
.downlevel .distill-header,
.downlevel .d-title,
.downlevel .d-abstract,
.downlevel .d-article,
.downlevel .d-appendix,
.downlevel .distill-appendix,
.downlevel .d-byline,
.downlevel .d-footnote-list,
.downlevel .d-citation-list,
.downlevel .distill-footer,
.downlevel .appendix-bottom,
.downlevel .posts-container {
  padding-left: 40px;
  padding-right: 40px;
}

@media(min-width: 768px) {
  .downlevel .base-grid,
  .downlevel .distill-header,
  .downlevel .d-title,
  .downlevel .d-abstract,
  .downlevel .d-article,
  .downlevel .d-appendix,
  .downlevel .distill-appendix,
  .downlevel .d-byline,
  .downlevel .d-footnote-list,
  .downlevel .d-citation-list,
  .downlevel .distill-footer,
  .downlevel .appendix-bottom,
  .downlevel .posts-container {
  padding-left: 150px;
  padding-right: 150px;
  max-width: 900px;
}
}

.downlevel pre code {
  display: block;
  border-left: 2px solid rgba(0, 0, 0, .1);
  padding: 0 0 0 20px;
  font-size: 14px;
}

.downlevel code, .downlevel pre {
  color: black;
  background: none;
  font-family: Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace;
  text-align: left;
  white-space: pre;
  word-spacing: normal;
  word-break: normal;
  word-wrap: normal;
  line-height: 1.5;

  -moz-tab-size: 4;
  -o-tab-size: 4;
  tab-size: 4;

  -webkit-hyphens: none;
  -moz-hyphens: none;
  -ms-hyphens: none;
  hyphens: none;
}

.downlevel .posts-list .post-preview {
  color: inherit;
}



</style>

<script type="application/javascript">

function is_downlevel_browser() {
  if (bowser.isUnsupportedBrowser({ msie: "12", msedge: "16"},
                                 window.navigator.userAgent)) {
    return true;
  } else {
    return window.load_distill_framework === undefined;
  }
}

// show body when load is complete
function on_load_complete() {

  // add anchors
  if (window.anchors) {
    window.anchors.options.placement = 'left';
    window.anchors.add('d-article > h2, d-article > h3, d-article > h4, d-article > h5');
  }


  // set body to visible
  document.body.style.visibility = 'visible';

  // force redraw for leaflet widgets
  if (window.HTMLWidgets) {
    var maps = window.HTMLWidgets.findAll(".leaflet");
    $.each(maps, function(i, el) {
      var map = this.getMap();
      map.invalidateSize();
      map.eachLayer(function(layer) {
        if (layer instanceof L.TileLayer)
          layer.redraw();
      });
    });
  }

  // trigger 'shown' so htmlwidgets resize
  $('d-article').trigger('shown');
}

function init_distill() {

  init_common();

  // create front matter
  var front_matter = $('<d-front-matter></d-front-matter>');
  $('#distill-front-matter').wrap(front_matter);

  // create d-title
  $('.d-title').changeElementType('d-title');

  // separator
  var separator = '<hr class="section-separator" style="clear: both"/>';
  // prepend separator above appendix
  $('.d-byline').before(separator);
  $('.d-article').before(separator);

  // create d-byline
  var byline = $('<d-byline></d-byline>');
  $('.d-byline').replaceWith(byline);

  // create d-article
  var article = $('<d-article></d-article>');
  $('.d-article').wrap(article).children().unwrap();

  // move posts container into article
  $('.posts-container').appendTo($('d-article'));

  // create d-appendix
  $('.d-appendix').changeElementType('d-appendix');

  // flag indicating that we have appendix items
  var appendix = $('.appendix-bottom').children('h3').length > 0;

  // replace footnotes with <d-footnote>
  $('.footnote-ref').each(function(i, val) {
    appendix = true;
    var href = $(this).attr('href');
    var id = href.replace('#', '');
    var fn = $('#' + id);
    var fn_p = $('#' + id + '>p');
    fn_p.find('.footnote-back').remove();
    var text = fn_p.html();
    var dtfn = $('<d-footnote></d-footnote>');
    dtfn.html(text);
    $(this).replaceWith(dtfn);
  });
  // remove footnotes
  $('.footnotes').remove();

  // move refs into #references-listing
  $('#references-listing').replaceWith($('#refs'));

  $('h1.appendix, h2.appendix').each(function(i, val) {
    $(this).changeElementType('h3');
  });
  $('h3.appendix').each(function(i, val) {
    var id = $(this).attr('id');
    $('.d-contents a[href="#' + id + '"]').parent().remove();
    appendix = true;
    $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('d-appendix'));
  });

  // show d-appendix if we have appendix content
  $("d-appendix").css('display', appendix ? 'grid' : 'none');

  // localize layout chunks to just output
  $('.layout-chunk').each(function(i, val) {

    // capture layout
    var layout = $(this).attr('data-layout');

    // apply layout to markdown level block elements
    var elements = $(this).children().not('details, div.sourceCode, pre, script');
    elements.each(function(i, el) {
      var layout_div = $('<div class="' + layout + '"></div>');
      if (layout_div.hasClass('shaded')) {
        var shaded_content = $('<div class="shaded-content"></div>');
        $(this).wrap(shaded_content);
        $(this).parent().wrap(layout_div);
      } else {
        $(this).wrap(layout_div);
      }
    });


    // unwrap the layout-chunk div
    $(this).children().unwrap();
  });

  // remove code block used to force  highlighting css
  $('.distill-force-highlighting-css').parent().remove();

  // remove empty line numbers inserted by pandoc when using a
  // custom syntax highlighting theme, except when numbering line
  // in code chunk
  $('pre:not(.numberLines) code.sourceCode a:empty').remove();

  // load distill framework
  load_distill_framework();

  // wait for window.distillRunlevel == 4 to do post processing
  function distill_post_process() {

    if (!window.distillRunlevel || window.distillRunlevel < 4)
      return;

    // hide author/affiliations entirely if we have no authors
    var front_matter = JSON.parse($("#distill-front-matter").html());
    var have_authors = front_matter.authors && front_matter.authors.length > 0;
    if (!have_authors)
      $('d-byline').addClass('hidden');

    // article with toc class
    $('.d-contents').parent().addClass('d-article-with-toc');

    // strip links that point to #
    $('.authors-affiliations').find('a[href="#"]').removeAttr('href');

    // add orcid ids
    $('.authors-affiliations').find('.author').each(function(i, el) {
      var orcid_id = front_matter.authors[i].orcidID;
      var author_name = front_matter.authors[i].author
      if (orcid_id) {
        var a = $('<a></a>');
        a.attr('href', 'https://orcid.org/' + orcid_id);
        var img = $('<img></img>');
        img.addClass('orcid-id');
        img.attr('alt', author_name ? 'ORCID ID for ' + author_name : 'ORCID ID');
        img.attr('src','data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==');
        a.append(img);
        $(this).append(a);
      }
    });

    // hide elements of author/affiliations grid that have no value
    function hide_byline_column(caption) {
      $('d-byline').find('h3:contains("' + caption + '")').parent().css('visibility', 'hidden');
    }

    // affiliations
    var have_affiliations = false;
    for (var i = 0; i<front_matter.authors.length; ++i) {
      var author = front_matter.authors[i];
      if (author.affiliation !== "&nbsp;") {
        have_affiliations = true;
        break;
      }
    }
    if (!have_affiliations)
      $('d-byline').find('h3:contains("Affiliations")').css('visibility', 'hidden');

    // published date
    if (!front_matter.publishedDate)
      hide_byline_column("Published");

    // document object identifier
    var doi = $('d-byline').find('h3:contains("DOI")');
    var doi_p = doi.next().empty();
    if (!front_matter.doi) {
      // if we have a citation and valid citationText then link to that
      if ($('#citation').length > 0 && front_matter.citationText) {
        doi.html('Citation');
        $('<a href="#citation"></a>')
          .text(front_matter.citationText)
          .appendTo(doi_p);
      } else {
        hide_byline_column("DOI");
      }
    } else {
      $('<a></a>')
         .attr('href', "https://doi.org/" + front_matter.doi)
         .html(front_matter.doi)
         .appendTo(doi_p);
    }

     // change plural form of authors/affiliations
    if (front_matter.authors.length === 1) {
      var grid = $('.authors-affiliations');
      grid.children('h3:contains("Authors")').text('Author');
      grid.children('h3:contains("Affiliations")').text('Affiliation');
    }

    // remove d-appendix and d-footnote-list local styles
    $('d-appendix > style:first-child').remove();
    $('d-footnote-list > style:first-child').remove();

    // move appendix-bottom entries to the bottom
    $('.appendix-bottom').appendTo('d-appendix').children().unwrap();
    $('.appendix-bottom').remove();

    // hoverable references
    $('span.citation[data-cites]').each(function() {
      const citeChild = $(this).children()[0]
      // Do not process if @xyz has been used without escaping and without bibliography activated
      // https://github.com/rstudio/distill/issues/466
      if (citeChild === undefined) return true

      if (citeChild.nodeName == "D-FOOTNOTE") {
        var fn = citeChild
        $(this).html(fn.shadowRoot.querySelector("sup"))
        $(this).id = fn.id
        fn.remove()
      }
      var refs = $(this).attr('data-cites').split(" ");
      var refHtml = refs.map(function(ref) {
        // Could use CSS.escape too here, we insure backward compatibility in navigator
        return "<p>" + $('div[id="ref-' + ref + '"]').html() + "</p>";
      }).join("\n");
      window.tippy(this, {
        allowHTML: true,
        content: refHtml,
        maxWidth: 500,
        interactive: true,
        interactiveBorder: 10,
        theme: 'light-border',
        placement: 'bottom-start'
      });
    });

    // fix footnotes in tables (#411)
    // replacing broken distill.pub feature
    $('table d-footnote').each(function() {
      // we replace internal showAtNode methode which is triggered when hovering a footnote
      this.hoverBox.showAtNode = function(node) {
        // ported from https://github.com/distillpub/template/pull/105/files
        calcOffset = function(elem) {
            let x = elem.offsetLeft;
            let y = elem.offsetTop;
            // Traverse upwards until an `absolute` element is found or `elem`
            // becomes null.
            while (elem = elem.offsetParent && elem.style.position != 'absolute') {
                x += elem.offsetLeft;
                y += elem.offsetTop;
            }

            return { left: x, top: y };
        }
        // https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetTop
        const bbox = node.getBoundingClientRect();
        const offset = calcOffset(node);
        this.show([offset.left + bbox.width, offset.top + bbox.height]);
      }
    })

    // clear polling timer
    clearInterval(tid);

    // show body now that everything is ready
    on_load_complete();
  }

  var tid = setInterval(distill_post_process, 50);
  distill_post_process();

}

function init_downlevel() {

  init_common();

   // insert hr after d-title
  $('.d-title').after($('<hr class="section-separator"/>'));

  // check if we have authors
  var front_matter = JSON.parse($("#distill-front-matter").html());
  var have_authors = front_matter.authors && front_matter.authors.length > 0;

  // manage byline/border
  if (!have_authors)
    $('.d-byline').remove();
  $('.d-byline').after($('<hr class="section-separator"/>'));
  $('.d-byline a').remove();

  // remove toc
  $('.d-contents').remove();

  // move appendix elements
  $('h1.appendix, h2.appendix').each(function(i, val) {
    $(this).changeElementType('h3');
  });
  $('h3.appendix').each(function(i, val) {
    $(this).nextUntil($('h1, h2, h3')).addBack().appendTo($('.d-appendix'));
  });


  // inject headers into references and footnotes
  var refs_header = $('<h3></h3>');
  refs_header.text('References');
  $('#refs').prepend(refs_header);

  var footnotes_header = $('<h3></h3');
  footnotes_header.text('Footnotes');
  $('.footnotes').children('hr').first().replaceWith(footnotes_header);

  // move appendix-bottom entries to the bottom
  $('.appendix-bottom').appendTo('.d-appendix').children().unwrap();
  $('.appendix-bottom').remove();

  // remove appendix if it's empty
  if ($('.d-appendix').children().length === 0)
    $('.d-appendix').remove();

  // prepend separator above appendix
  $('.d-appendix').before($('<hr class="section-separator" style="clear: both"/>'));

  // trim code
  $('pre>code').each(function(i, val) {
    $(this).html($.trim($(this).html()));
  });

  // move posts-container right before article
  $('.posts-container').insertBefore($('.d-article'));

  $('body').addClass('downlevel');

  on_load_complete();
}


function init_common() {

  // jquery plugin to change element types
  (function($) {
    $.fn.changeElementType = function(newType) {
      var attrs = {};

      $.each(this[0].attributes, function(idx, attr) {
        attrs[attr.nodeName] = attr.nodeValue;
      });

      this.replaceWith(function() {
        return $("<" + newType + "/>", attrs).append($(this).contents());
      });
    };
  })(jQuery);

  // prevent underline for linked images
  $('a > img').parent().css({'border-bottom' : 'none'});

  // mark non-body figures created by knitr chunks as 100% width
  $('.layout-chunk').each(function(i, val) {
    var figures = $(this).find('img, .html-widget');
    // ignore leaflet img layers (#106)
    figures = figures.filter(':not(img[class*="leaflet"])')
    if ($(this).attr('data-layout') !== "l-body") {
      figures.css('width', '100%');
    } else {
      figures.css('max-width', '100%');
      figures.filter("[width]").each(function(i, val) {
        var fig = $(this);
        fig.css('width', fig.attr('width') + 'px');
      });

    }
  });

  // auto-append index.html to post-preview links in file: protocol
  // and in rstudio ide preview
  $('.post-preview').each(function(i, val) {
    if (window.location.protocol === "file:")
      $(this).attr('href', $(this).attr('href') + "index.html");
  });

  // get rid of index.html references in header
  if (window.location.protocol !== "file:") {
    $('.distill-site-header a[href]').each(function(i,val) {
      $(this).attr('href', $(this).attr('href').replace(/^index[.]html/, "./"));
    });
  }

  // add class to pandoc style tables
  $('tr.header').parent('thead').parent('table').addClass('pandoc-table');
  $('.kable-table').children('table').addClass('pandoc-table');

  // add figcaption style to table captions
  $('caption').parent('table').addClass("figcaption");

  // initialize posts list
  if (window.init_posts_list)
    window.init_posts_list();

  // implmement disqus comment link
  $('.disqus-comment-count').click(function() {
    window.headroom_prevent_pin = true;
    $('#disqus_thread').toggleClass('hidden');
    if (!$('#disqus_thread').hasClass('hidden')) {
      var offset = $(this).offset();
      $(window).resize();
      $('html, body').animate({
        scrollTop: offset.top - 35
      });
    }
  });
}

document.addEventListener('DOMContentLoaded', function() {
  if (is_downlevel_browser())
    init_downlevel();
  else
    window.addEventListener('WebComponentsReady', init_distill);
});

</script>

<!--/radix_placeholder_distill-->
  <script src="../../site_libs/header-attrs-2.26/header-attrs.js"></script>
  <script src="../../site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
  <script src="../../site_libs/popper-2.6.0/popper.min.js"></script>
  <link href="../../site_libs/tippy-6.2.7/tippy.css" rel="stylesheet" />
  <link href="../../site_libs/tippy-6.2.7/tippy-light-border.css" rel="stylesheet" />
  <script src="../../site_libs/tippy-6.2.7/tippy.umd.min.js"></script>
  <script src="../../site_libs/anchor-4.2.2/anchor.min.js"></script>
  <script src="../../site_libs/bowser-1.9.3/bowser.min.js"></script>
  <script src="../../site_libs/webcomponents-2.0.0/webcomponents.js"></script>
  <script src="../../site_libs/distill-2.2.21/template.v2.js"></script>
  <!--radix_placeholder_site_in_header-->
<style type="text/css">
.distill-site-nav {
  color: #000000;
  background-color: #EFEFEF;
  font-weight: 200;
  font-size: 13pt;
}

.distill-site-nav a {
  color: inherit;
  text-decoration: none;
}

.distill-site-nav a:hover {
  color: black;
}

.distill-site-header {
}

.distill-site-footer {
}

@media print {
  .distill-site-nav {
    display: none;
  }
}

body {
  max-width: 1100px;
  margin: auto;
 padding: 1em;
 line-height: 20px ; 
}
</style>
<!--/radix_placeholder_site_in_header-->


</head>

<body>

<!--radix_placeholder_front_matter-->

<script id="distill-front-matter" type="text/json">
{"title":"Genome-wide association studies","description":"An overview of genome-wide association studies.","authors":[{"author":"Tim Barry","authorURL":"https://timothy-barry.github.io","affiliation":"&nbsp;","affiliationURL":"#","orcidID":""}],"publishedDate":"2024-05-01T00:00:00.000-04:00","citationText":"Barry, 2024"}
</script>

<!--/radix_placeholder_front_matter-->
<!--radix_placeholder_navigation_before_body-->
<header class="header header--fixed" role="banner">
<nav class="distill-site-nav distill-site-header">
<div class="nav-left">
<a href="../../index.html" class="title">Tim Barry</a>
</div>
<div class="nav-right">
<a href="../../research.html">Research</a>
<a href="../../blog.html">Blog</a>
<a href="../../fun.html">Fun</a>
<a href="javascript:void(0);" class="nav-toggle">&#9776;</a>
</div>
</nav>
</header>
<!--/radix_placeholder_navigation_before_body-->
<!--radix_placeholder_site_before_body-->
<!--/radix_placeholder_site_before_body-->

<div class="d-title">
<h1>Genome-wide association studies</h1>

<!--radix_placeholder_categories-->
<div class="dt-tags">
  <a href="../../blog.html#category:Genomics" class="dt-tag">Genomics</a>
</div>
<!--/radix_placeholder_categories-->
<p><p>An overview of genome-wide association studies.</p></p>
</div>

<div class="d-byline">
  Tim Barry <a href="https://timothy-barry.github.io" class="uri">https://timothy-barry.github.io</a> 
  
<br/>05-01-2024
</div>

<div class="d-article">
<p>After years of observing from the outside, I am getting into the genetics space (more on that later). This post is an overview of the most fundamental of topics in genetics: genome-wide association studies (GWAS). First, I provide some background on molecular genetics. Then, I discuss the methodology underlying GWAS. Finally, I conclude with a discussion of the successes and limitations of GWAS.</p>
<h2 id="molecular-genetics-background">Molecular genetics background</h2>
<p>The human genome consists of a set of 6 billion letters (or <em>nucleotides</em>). Each nucleotide can take one of four values: A, T, G, or C. An individual’s set of nucleotides can be arranged into two non-overlapping “strings” of nucleotides. One of these nucleotide strings is inherited from one’s mother, and the other is inherited from one’s father. The strings are “homologous” in the sense that they are the same length (to first approximation) and can be thought of as a pair. The strings can be arranged linearly side-by-side. For example, in the example genome depicted in Figure 1, there is an “A” nucleotide (inherited from the mom) and a “G” nucleotide (inherited from the dad) at the second position from the left. We call a continuous region of the genome a <em>locus</em>. A locus can consist of a single nucleotide or a range of contiguous nucleotides.</p>
<p>The genome is partitioned into distinct physical pieces called chromosomes; there are 22 non-sex related chromosomes, or <em>autosomes</em>, and two sex-related chromosomes in the human genome. (We focus only on the autosomes in this blog post.) The genomic coordinate of a given locus is specified by the chromosome of the locus and the position along the chromosome at which the locus is located. For example, “chr1:150000” refers to the nucleotide on chromosome 1 located at position 150,000. Note that the notation “chr1:150000” does not refer specifically to either the maternal chromosome (i.e., the chromosome inherited from the mom) or the paternal chromosome (i.e., the chromosome inherited from the dad).</p>
<br/>
<center>
<figure id="id">
<img src="genome_1.png" class="class" style="width:100.0%;height:100.0%" alt="Figure 1: Cartoon of an individual’s genome" />
<figcaption aria-hidden="true">Figure 1: Cartoon of an individual’s genome</figcaption>
</figure>
</center>
<p>At a given single-nucleotide locus, there may or may not be variation in the population. For example, all individuals in the population may have a “G” nucleotide on both their maternal and paternal chromosome at a given position. (That is to say, <em>all</em> chromosomes in a population may have the same nucleotide at one position.) On the other hand, at certain locations in the genome, there may be variation across the chromosomes of a population. For example, at a given single-nucleotide locus, half of chromosomes may have an “A” nucleotide and the other half of chromosomes may have a “C” nucleotide. Single-nucleotide loci for which there is variation across chromosomes are called <em>single-nucleotide polymorphisms</em> (or SNPs). The different nucleotides present at a given SNP are called <em>alleles</em>. We typically assume that a given SNP is <em>biallelic</em>, i.e., that it comes in two versions, for example “A” or “C.” We label one of the alleles the <em>reference</em> allele and the other allele the <em>alternative</em> allele; this labeling is arbitrary. If an individual has two copies of the same allele (at a given location), we say that the individual is <em>homozygous</em> (at that location); by contrast, if an individual has one copy of the reference allele and one copy of the alternative allele, we say that the individual is <em>heterozygous</em> (at that location).</p>
<p>The goal of a <em>genome-wide association study (GWAS)</em> is to identify single-nucleotide loci that <em>cause</em> (or, less ambitiously, are <em>associated with</em>) a given trait or disease of interest. In carrying out a GWAS, we restrict our attention to SNPs, as only single-nucleotide loci that vary across individuals explain variation in a trait or disease. <em>Neutral theory</em> predicts that most SNPs in a given population are neutral with respect to a given trait, i.e. they do not affect (or explain variation) in the trait. Thus, a GWAS is a high-dimensional variable selection problem: we seek to identify a (relatively small) number of causal SNPs among a large set of neutral SNPs.</p>
<p>A <em>DNA (or SNP) microarray</em> is a device that enables the measurement of a predefined set of 500,000 — 1,000,000 SNPs. One can use a DNA microarray to measure each of <span class="math inline">\(\approx 500,000\)</span> SNPs in a large number (e.g., <span class="math inline">\(\approx 500,000\)</span>) of individuals. More recently, <em>whole exome sequencing</em> and <em>whole genome sequencing</em> have enabled the measurement of the <em>exome</em> (i.e., the set of coding regions of the genome) and the entire genome, respectively. Whole exome sequencing and whole genome sequencing data are collected using a <em>next-generation DNA sequencing machine</em> (instead of a DNA microarray). Our focus in this blog post is on DNA microarray data, as such data currently are more widespread than whole exome or whole genome sequencing data. (However, whole exome sequencing and whole genome sequencing data quickly are increasing in prevalence.)</p>
<h2 id="sexual-reproduction-recombination-and-linkage-disequilibrium">Sexual reproduction, recombination, and linkage disequilibrium</h2>
<p>Sexual reproduction is an important topic in the context of GWAS, as the mechanics of sexual reproduction help explain the correlation structure of a set of SNPs. A <em>diploid cell</em> is a cell that contains 46 chromosomes (i.e., 22 pairs of autosomes and two sex chromosomes). Next, a <em>haploid cell</em> is a cell that contains only 23 chromosomes (i.e., a single set of 22 autosomes and a single sex chromosome). <em>Somatic</em> cells (i.e., body cells) are diploid cells, whereas <em>germ cells</em> (i.e., eggs and sperm) are haploid cells. To first approximation, sexual reproduction works as follows: within both the male partner and the female partner, a (diploid) somatic cell divides to form “daugther” (haploid) germ cells. Then, the male and female germ cells combine to form a diploid <em>zygote</em> in a process called <em>fertilization</em>. The zygote then multiplies via cell division, becoming an embryo and then a fetus, etc.</p>
<p>The process by which a diploid cell divides to form haploid daughter cells is called <em>meoisis</em>. Meoisis is a complicated cellular processes, but from our perspective, the most important event in meoisis is <em>recombination</em>. During recombination, corresponding segments of a pair of homologous chromosomes are exchanged to form a new “daughter chromosome.” The daughter chromosome is a mix of the maternal and paternal chromosomes. Figure 2 illustrates recombination of a given chromosome (e.g., chromosome 1) during meoisis. Note that recombination occurs in both the male and female partners before fertilization. Meoisis helps increase the genetic diversity of an individual’s (multiple) offspring, thereby conferring a selective advantage to the parent.</p>
<p>As a result of genetic recombination, SNPs that are in close physical proximity tend to be inherited together, while SNPs that are far away (or on different chromosomes) tend to be inherited (approximately) independently of one another. This phenomenon — i.e., the phenomenon by which nearby SNPs are inherited together — is called <em>linkage disequilibrium (LD)</em>. An <em>LD block</em> is a segment of the genome in which there is minimal genetic recombination. LD has important implications for the correlation structure of SNPs, as we will see.</p>
<br/>
<center>
<figure id="id">
<img src="sex_repro.png" class="class" style="width:100.0%;height:100.0%" alt="Figure 2: Sexual reproduction and recombination" />
<figcaption aria-hidden="true">Figure 2: Sexual reproduction and recombination</figcaption>
</figure>
</center>
<h2 id="hardy-weinberg-equilibrium">Hardy-Weinberg equilibrium</h2>
<p>Consider an individual and a SNP in the population. Let <span class="math inline">\(I_1 \in \{0,1\}\)</span> (resp., <span class="math inline">\(I_2 \in \{0,1\}\)</span>) be a binary variable that takes value 1 if the maternal chromosome (resp., paternal chromosome) contains the alternative allele (and 0 otherwise). The number of alternative alleles <span class="math inline">\(X\)</span> that the individual possesses is equal to the sum of <span class="math inline">\(I_1\)</span> and <span class="math inline">\(I_2\)</span>, i.e. <span class="math display">\[X = I_1 + I_2.\]</span> If <span class="math inline">\(I_1\)</span> and <span class="math inline">\(I_2\)</span> are i.i.d. <span class="math inline">\(\textrm{Bern}(p)\)</span> random variables, then <span class="math inline">\(X \sim \textrm{binom}(2,p)\)</span>. <span class="math inline">\(I_1\)</span> and <span class="math inline">\(I_2\)</span> are i.i.d. (and thus <span class="math inline">\(X\)</span> is binomially distributed) when (i) the population is infinitely large and (ii) mating among individuals in the population is random. If <span class="math inline">\(X\)</span> is binomially distributed, then we say that the corresponding SNP follows <em>Hardy-Weinberg equilibrium</em>. In practice, one can test if <span class="math inline">\(X\)</span> is binomially distributed by applying a binomial goodness of fit test. Most SNPs in the human genome turn out to be in approximate Hardy-Weinberg equilibrium. SNPs that fail to satisfy Hardy-Weinberg equilibrium generally are removed from GWAS, as departure from Hardy-Weinberg equilibrium typically results from technical errors in the genotyping step.</p>
<h2 id="snp-matrix-common-and-rare-variants-correlation-structure-among-snps">SNP matrix, common and rare variants, correlation structure among SNPs</h2>
<p>Suppose we use a SNP array to measure each of <span class="math inline">\(p\)</span> SNPs on a set of <span class="math inline">\(n\)</span> individuals. We obtain a <em>SNP matrix</em> <span class="math inline">\(M \in \{0, 1, 2\}^{n \times p}\)</span>. Each entry of the SNP matrix takes a value in the set <span class="math inline">\(\{0, 1, 2\}\)</span>; by convention, the <span class="math inline">\(n\)</span> individuals are stored in the rows and the <span class="math inline">\(p\)</span> SNPs are stored in the columns. A given entry <span class="math inline">\(X_{i,j}\)</span> of the matrix indicates the number of alternative alleles measured for individual <span class="math inline">\(i\)</span> at SNP <span class="math inline">\(j\)</span>. (The specific reference and alternative alleles at SNP <span class="math inline">\(j\)</span> — for example, “A” and “T” — are not important.) A 0 indicates homozygosity of the reference allele; a 1 indicates heterozygosity; and a 2 indicates homozygosity of the alternative allele.</p>
<p>Consider a given biallelic SNP. The <em>major allele</em> (resp, <em>minor allele</em>) is the allele that occurs more (resp., less) frequently in the population. (Note that the major and minor alleles are conceptually distinct from the reference and alternative alleles; in other words, the major allele could be <em>either</em> the reference allele <em>or</em> the alternative allele, and the same goes for the minor allele.) The <em>minor allele frequency</em> is the frequency of the minor allele in the population. We can use the SNP matrix to estimate the MAF of a given SNP. Let <span class="math inline">\(X_1, \dots, X_n\)</span> denote the entries of the <span class="math inline">\(j\)</span>th column of <span class="math inline">\(M\)</span> (i.e., the column of <span class="math inline">\(M\)</span> corresponding to SNP <span class="math inline">\(j\)</span>). An estimate for the minor allele frequency is <span class="math display">\[\widehat{\textrm{MAF}} = \min \left\{\frac{\sum_{i=1}^n X_i}{2n}, 1 -  \frac{\sum_{i=1}^n X_i}{2n} \right\}.\]</span> (Observe that <span class="math inline">\(\sum_{i=1}^n X_i\)</span> is the number of copies of the alternative allele across individuals, while <span class="math inline">\(2n\)</span> is the total number of alleles.)</p>
<br/>
<center>
<figure id="id">
<img src="snp_matrix.png" class="class" style="width:80.0%" alt="Figure 3: An example SNP matrix" />
<figcaption aria-hidden="true">Figure 3: An example SNP matrix</figcaption>
</figure>
</center>
<p>A <em>rare variant</em> is a SNP whose minor allele frequency is small, where “small” operationally is defined as <span class="math inline">\(1\%\)</span>. By contrast, a <em>common variant</em> is a SNP that is not a rare variant. Typically, we filter out rare variants in a GWAS, as the “effective sample size” of rare variants is insufficiently large to enable well-powered inference. However, as GWAS sample sizes increase, rare variant analysis is becoming increasingly practical. Rare variants typically exert a larger effect on the phenotype than common variants (Figure 4). In other words, as the MAF of a SNP increases, its effect size on the phenotype generally decreases. This phenomenon occurs because natural selection tends to remove alleles with large, deleterious effects from the gene pool. Rare variants are especially interesting in the context of drug discovery, as their alteration potentially could lead to especially large changes to the phenotype. A special type of rare variant is a <em>de novo variant</em>. A de novo variant is a variant that appears for the first time in the germ cell of a parent (and is not present in the somatic cells of the parent). De novo variants result from mutations in a sperm or egg cell.</p>
<br/>
<center>
<figure id="id">
<img src="size_vs_maf.png" class="class" style="width:60.0%" alt="Figure 4: Common vs. rare variant effect sizes" />
<figcaption aria-hidden="true">Figure 4: Common vs. rare variant effect sizes</figcaption>
</figure>
</center>
<p>Due to linkage disequilibrium, SNPs that are in close physical proximity tend to be highly correlated with one another (across individuals in a population). Let <span class="math inline">\(\Sigma = (1/n) X X^T\)</span> denote the empirical correlation matrix of the SNPs. Let <span class="math inline">\(\Sigma_{i,j}\)</span> denote the correlation between SNPs <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span>. A very rough model for <span class="math inline">\(\Sigma\)</span> is as follows. If SNPs <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> are in the same LD block, then <span class="math inline">\(\Sigma_{i,j}\)</span> is big (e.g., <span class="math inline">\(\Sigma_{i,j} \approx 0.99\)</span>); on the other hand, if SNPs <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> are in different LD blocks, then <span class="math inline">\(\Sigma_{i,j}\)</span> is fairly small (e.g., <span class="math inline">\(\Sigma_{i,j} \approx 0.15\)</span>). (SNPs in different LD blocks tend to be weakly but positively correlated due to population stratification, which we will discuss shortly.)</p>
<h2 id="gwas">GWAS</h2>
<p>Genome wide association studies (GWAS) aim to identify SNPs associated with a given disease or trait. A GWAS proceeds as follows. First, we select a trait of interest. The trait can be binary (e.g., presence or absence of bipolar disorder) or continuous (e.g., blood lipid level). For simplicity we will assume that the trait is a binary disease status (although the analysis proceeds similarly if the trait is continuous). Next, we collect data on <span class="math inline">\(n_1\)</span> subjects <em>with</em> the disease and <span class="math inline">\(n_2\)</span> healthy controls. For each of the <span class="math inline">\(n := n_1 + n_2\)</span> subjects, we measure <span class="math inline">\(p \approx 500,000\)</span> SNPs, resulting in a SNP matrix <span class="math inline">\(X \in \{0,1,2\}^{n \times p}\)</span>. Additionally, we collect demographic factors <span class="math inline">\(Z_1, \dots, Z_n \in \mathbb{R}^d\)</span> on each subject, such as age and sex. We assume that all subjects come from the same “ancestry group.” (As far as I can tell, there are three main “ancestry groups” used in modern genetics research: African, Asian, and European.) Finally, let <span class="math inline">\(Y = [Y_1, \dots, Y_n]^T \in \{0,1\}^n\)</span> denote the binary vector indicating whether a given individual has the disease.</p>
<p>There are five main steps involved in carrying out a GWAS: performing quality control on the SNP matrix, imputing missing SNPs, performing PCA on the SNP matrix, carrying out the regressions, and then applying a multiple testing correction. We describe each of these steps one-by-one.</p>
<h3 id="quality-control-qc">Quality control (QC)</h3>
<p>In practice the SNP matrix typically does not look quite like the SNP matrix depicted in Figure 3. Rather, some of the entries are missing due to technical errors in the genotyping process. A more realistic SNP matrix is shown in Figure 5. A subset of the entries is missing, as denoted by “NA.” The quality control step involves removing low-quality SNPs (columns) and subjects (rows) from the analysis. (Not all quality control steps relate to missingness.)</p>
<br/>
<center>
<figure id="id">
<img src="snp_matrix_2.png" class="class" style="width:80.0%" alt="Figure 5: Example of a SNP matrix with missingness." />
<figcaption aria-hidden="true">Figure 5: Example of a SNP matrix with missingness.</figcaption>
</figure>
</center>
<ul>
<li><strong>SNPs and subjects with many missing values</strong>. Some SNPs and subjects may have a large proportion of missing values. A common practice is to exclude SNPs and subjects whose missingness exceeds 10%.</li>
<li><strong>Heterozygosity check</strong>. If a given subject is heterozygous on most alleles (i.e., the subject has a very large fraction of 1’s and a comparatively small fraction of 0’s and 2’s), there likely were technical problems in genotyping that subject. Subjects with very high heterozygosity typically are removed from the analysis.</li>
<li><strong>Minor allele frequency filtering</strong>. Recall that we can estimate the minor allele frequency of a given SNP via <span class="math inline">\(\widehat{MAF}\)</span>. SNPs with a very low minor allele frequency have a small “effective sample size” and thus are challenging to perform statistical inference on. We typically remove SNPs whose minor allele frequency is below 10%.</li>
<li><strong>Hardy-Weinberg equilibrium filtering</strong>. Recall that most SNPs in the population follow Hardy-Weinberg equilibrium, i.e. the number of alternative alleles that a given subject possesses is approximately binomially distributed. SNPs that deviate from Hardy-Weinberg equilibrium likely were measured incorrectly and thus are removed. We can check for Hardy-Weinberg equilibrium by running a binomial goodness of fit test on the data for each SNP.</li>
<li><strong>Additional QC checks</strong>. Additional QC checks sometimes are performed, such as checking for relatedness of subjects to ensure (approximate) independence across samples.</li>
</ul>
<h3 id="snp-imputation">SNP imputation</h3>
<p>Even after performing QC, some entries of the SNP matrix will be missing. In preparation for the next step (i.e., PCA on the SNP matrix), it is standard to impute the missing entries of the SNP matrix. Both simple and sophisticated imputation techniques have been developed for this purpose. One common technique is to assign the most common value (i.e., 0, 1, or 2) of a given SNP to the missing entries of that SNP. This strategy, although simple, seems to work quite well in practice.</p>
<h3 id="pca-on-the-snp-matrix">PCA on the SNP matrix</h3>
<p><em>Population stratification</em> refers to the presence of multiple distinct “subpopulations” in a collection of subjects. Population stratification can exert a confounding effect, leading to spurious associations and a loss of type-I error control. For example, suppose that a given sample consists of two subpopulations, namely “subpopulation A” and “subpopulation B.” Consider a given SNP. Suppose that the alternative allele is more prevalent in subpopulation A than B. Moreover, suppose that members of subpopulation A are more likely than members of subpopulation B to have the disease. Suppose further that the increased prevalence of the disease in subpopulation A is <em>not</em> due to the given SNP. (For example, members of subpopulation A may live in a different environment than members of subpopulation B, and the disease may have an environmental component.) If we naively tested for association between the given SNP and presence or absence of the disease, we would incorrectly conclude that the SNP is associated with the disease. It is therefore necessary to control for subpopulation to neutralize this confounding effect.</p>
<p>The standard approach to adjust for population stratification is to perform PCA on the SNP matrix. The PCA yields <span class="math inline">\(k\)</span> PCs <span class="math inline">\(v_1, \dots, v_n \in \mathbb{R}^n\)</span> (where <span class="math inline">\(k\)</span> is chosen heuristically or set to some prespecified number, e.g. <span class="math inline">\(\approx 15\)</span>). The PCs capture population heterogeneity and can be included in the regression model to control for population stratification.</p>
<h3 id="regression">Regression</h3>
<p>The next (and in my opinion most interesting) step of the GWAS analysis pipeline is to test for association between each SNP and the phenotype by fitting a large number of logistic regression models. Consider a given SNP. For a given subject <span class="math inline">\(i\)</span>, let <span class="math inline">\(y_i \in \{0,1\}\)</span> denote the phenotype; let <span class="math inline">\(x_i \in \{0,1,2\}\)</span> denote the number of alternative alleles; let <span class="math inline">\(z_i \in \mathbb{R}^d\)</span> denote the vector of demographic factors (e.g., age, sex, etc.); and let <span class="math inline">\(w_i \in \mathbb{R}^k\)</span> denote the top <span class="math inline">\(k\)</span> PCs. We hypothesize the following logistic regression model for the data: <span class="math display">\[y_i \sim \textrm{Bern}(\mu_i); \quad \textrm{logit}(\mu_i) = \beta_0 + \beta x_i + \delta^T w_i + \gamma^T z_i,\]</span> where <span class="math inline">\(\beta_0 \in \mathbb{R}, \beta \in \mathbb{R}, \delta \in \mathbb{R}^k\)</span> and <span class="math inline">\(\gamma \in \mathbb{R}^d\)</span> are unknown regression coefficients.</p>
<p>We fit the model via MLE, yielding an estimate <span class="math inline">\(\hat{\beta}\)</span> for <span class="math inline">\(\beta\)</span> and a p-value <span class="math inline">\(p\)</span> for a test of the null hypothesis <span class="math inline">\(H_0: \beta = 0\)</span>. (This p-value typically is obtained via Wald or likelihood ratio test.) If the trait is continuous, then we fit a linear regression model rather than a logistic regression model to the data.</p>
<h3 id="multiple-testing-correction">Multiple testing correction</h3>
<p>The regression analysis yields a set of <span class="math inline">\(p\)</span> p-values <span class="math inline">\(p_1, \dots, p_p\)</span>. We obtain a discovery set via a very simple procedure: we call all SNPs whose p-value falls below the threshold of <span class="math inline">\(5 \cdot 10^{-8}\)</span> as discoveries. Some more recent GWAS analyses apply a BH correction to the p-values to produce a discovery set, but this does not appear to be the norm.</p>
<h2 id="takeaways-and-limitations-of-gwas">Takeaways and limitations of GWAS</h2>
<p>GWAS have identified hundreds of thousands of SNPs associated with hundreds of traits and diseases, yielding several interesting (and perhaps surprising) insights about the architecture of complex traits. For example, most traits are highly polygenic, i.e., they are influenced by tens of thousands (or more) of SNPs spread aross the genome. Additionally, the vast majority (<span class="math inline">\(&gt;95\%\)</span>) of significant SNPs lie in noncoding regions of the genome. These noncoding SNPs are thought to contribute to disease by modulating the expression of one more or disease-relevant genes.</p>
<p>Despite their success, GWAS present several limitations, two of which we highlight here. First, due to LD (i.e., multicolinearity among the SNPs), it is challenging to identify a causal SNP among a set of SNPs in close physical proximity to one another. Second, it is unclear <em>how</em> SNPs in noncoding regions in the genome impact the phenotype. In particular, the gene (or genes) through which most noncoding SNP exert their effect is unknown. These limitations have impeded the translation of GWAS results into therapeutics for genetic diseases. In future blog posts I will explore genomic technologies (e.g., eQTLs studies, CRISPR experiments) and analytic tools that have been developed to overcome the limitations of GWAS.</p>
<h2 id="references">References</h2>
<p>This post is based on a couple excellent references on GWAS.</p>
<ul>
<li><p>The <a href="https://www.mv.helsinki.fi/home/mjxpirin/GWAS_course/">GWAS lecture notes</a> by Prof Matti Pirinen.</p></li>
<li><p>The <a href="https://pbreheny.github.io/adv-gwas-tutorial/">GWAS tutorial</a> by Prof Patrick Breheny.</p></li>
</ul>
<div class="sourceCode" id="cb1"><pre class="sourceCode r distill-force-highlighting-css"><code class="sourceCode r"></code></pre></div>
<!--radix_placeholder_article_footer-->
<!--/radix_placeholder_article_footer-->
</div>

<div class="d-appendix">
</div>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

<!--radix_placeholder_site_after_body-->
<!--/radix_placeholder_site_after_body-->
<!--radix_placeholder_appendices-->
<div class="appendix-bottom"></div>
<!--/radix_placeholder_appendices-->
<!--radix_placeholder_navigation_after_body-->
<!--/radix_placeholder_navigation_after_body-->

</body>

</html>
