%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Medium Length Professional CV
% LaTeX Template
% Version 2.0 (8/5/13)
%
% This template has been downloaded from:
% http://www.LaTeXTemplates.com
%
% Original author:
% Trey Hunner (http://www.treyhunner.com/)
%
% Important note:
% This template requires the resume.cls file to be in the same directory as the
% .tex file. The resume.cls file provides the resume style used for structuring the
% document.
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%----------------------------------------------------------------------------------------
%	PACKAGES AND OTHER DOCUMENT CONFIGURATIONS
%----------------------------------------------------------------------------------------

\documentclass{resume} % Use the custom resume.cls style
\usepackage{array}
\newcolumntype{L}[1]{>{\raggedright\let\newline\\\arraybackslash\hspace{0pt}}m{#1}}
\newcolumntype{C}[1]{>{\centering\let\newline\\\arraybackslash\hspace{0pt}}m{#1}}
\newcolumntype{R}[1]{>{\raggedleft\let\newline\\\arraybackslash\hspace{0pt}}m{#1}}
\usepackage[left=0.75in,top=0.6in,right=0.75in,bottom=0.6in]{geometry} % Document margins
\usepackage{xcolor}
\usepackage{hyperref}
\hypersetup{
	colorlinks=true,
	linkcolor=blue,
	filecolor=magenta,      
	urlcolor=MidnightBlue,
}

 \renewcommand{\familydefault}{\sfdefault}

\name{Timothy Barry} % Your name
\address{\texttt{tbarry2@andrew.cmu.edu} $\bullet$ \texttt{https://timothy-barry.github.io}} % Your address


\begin{document}

%----------------------------------------------------------------------------------------
%	EDUCATION SECTION
%----------------------------------------------------------------------------------------

\begin{rSection}{EDUCATION}

\begin{tabular}{p{14cm}R{2.5cm}}
	\bf{Carnegie Mellon University}  &  \it{2018 -- 2023}  \\ 
	PhD in Statistics and Data Science &  \\ Advisors: Kathryn Roeder, Eugene Katsevich  & \\ &\\
	\textbf{University of Maryland, College Park} & \it{2014 -- 2018}
	 \\  BS in Mathematics with high honors & \\
	 Minor in Computer Science
\end{tabular} 
\end{rSection}



%----------------------------------------------------------------------------------------
%	WORK EXPERIENCE SECTION
%----------------------------------------------------------------------------------------

% \begin{rSection}{INTERNSHIPS AND EXPERIENCE}

% \begin{tabular}{p{14cm}R{2.5cm}}
%	\textbf{Machine Learning Team, National Institutes of Health}  &  \it{Summer 2018} \\ 
%	$\bullet$ Trained statistical machine learning algorithms to forecast the recovery trajectory of individuals who suffered stroke. & \\$\bullet$ Advisors: Francisco Pereira  \& Charles Zheng & \\ & \\
%	\textbf{Center for Quantitative Medicine, UConn Health} & \textit{Summer 2017}\\
%	$\bullet$ Constructed a mathematical model of iron metabolism in the human body; used the model to test \textit{in silico} the effect of treatments on iron-abnormal individuals.  & \\ $\bullet$ Advisor: Pedro Mendes & \\ & \\
%	\textbf{Laboratory of William Fagan, University of Maryland} & \it{2015 -- 2018} \\ 
%	$\bullet$ Investigated wolf interaction with humans and human infrastructure using a general linear and mixed-effects modeling framework.  &  \\ $\bullet$  Advisor: Eliezer Gurarie & \\ 
% \end{tabular} 


% \end{rSection}

%---------------------------------------------------------------------------------------
%	Honors and awards 
%----------------------------------------------------------------------------------------

\begin{rSection}{AWARDS}


\begin{tabular}{p{14cm}R{2.5cm}}
$\bullet$ Howard Hughes Medical Institute Fellowship  &  \it{2017}  \\ 
$\bullet$ Maryland Summer Scholars Research Grant  & \it{2016}  \\ 
$\bullet$ Banneker-Key Scholarship, University of Maryland's most prestigious scholarship & \it{2014}
\end{tabular} 

% \begin{tabular}{ @{} >{\bfseries}l @{\hspace{6ex}} l }
%Computer Languages & Prolog, Haskell, AWK, Erlang, Scheme, ML \\
%Protocols \& APIs & XML, JSON, SOAP, REST \\
%Databases & MySQL, PostgreSQL, Microsoft SQL \\
%Tools & SVN, Vim, Emacs
% \end{tabular}

\end{rSection}

% Computing


\begin{rSection}{TEACHING ASSISTANTSHIPS}
\begin{tabular}{p{14cm}R{2.5cm}}
	$\bullet$ Statistics 36-350: Statistical computing 
	& \textit{Fall 2018} \\ 
	$\bullet$ Statistics 36-469: Statistical genomics and high-dimensional inference & \textit{Spring 2020}
\end{tabular} 
\end{rSection}

\begin{rSection}{MENTORING}
	\begin{tabular}{p{14cm}R{2.5cm}}
		$\bullet$ Songcheng Dai (Computational Biology Masters student at CMU). Research topic: algorithmic and statistical tools for large-scale single-cell data.
		& \textit{2021 - 2022} \\ 
	\end{tabular} 
\end{rSection}

\begin{rSection}{PAPERS}

\begin{tabular}{p{14cm}R{2.5cm}}
	$\bullet$ \textbf{T Barry}, E Katsevich, K Roeder. ``Robust differential expression testing for single-cell CRISPR screens.'' Preprint. Link.
	
	$\bullet$ \textbf{T Barry}, E Katsevich, K Roeder. “Exponential family measurement error models for single-cell CRISPR screens.” Revisions at Biostatistics. \href{https://arxiv.org/abs/2201.01879}{Link}. & \textit{2022} \\
	 
	$\bullet$ \textbf{T Barry}, X Wang, J Morris, K Roeder, E Katsevich. ``SCEPTRE improves calibration and sensitivity in single-cell CRISPR screen analysis.'' Genome Biology. \href{https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02545-2}{Link}. \textcolor{blue}{(Reviewers’ choice, American Society of Human Genetics conference, 2021)} 
	& \textit{2021}  
	\\
	$\bullet$ J Morris, Z Daniloski, J Domingo, \textbf{T Barry}, M Ziosi, D Glinos, S Hao, E Mimitou, P Smibert, K Roeder, E Katsevich, T Lappalainen, N Sanjana. “Discovery of target genes and pathways of blood trait loci using pooled CRISPR screens and single cell RNA sequencing.” BioRxiv. \href{https://www.biorxiv.org/content/10.1101/2021.04.07.438882v1}{Link}.  & \textit{2021} \\
\end{tabular} 

\textbf{Undergraduate}

\begin{tabular}{p{14cm}R{2.5cm}}
	$\bullet$ \textbf{T Barry}*, E Gurarie*, F Cheraghi, I Kajola, W Fagan. ``Does dispersal make the heart grow bolder? Avoidance of anthropogenic habitat elements across wolf life history.'' \textit{Animal Behaviour} 166. *\textit{Joint first authorship}. \href{https://www.sciencedirect.com/science/article/pii/S0003347220301743}{Link}. & \textit{2020}  \\
	$\bullet$ \textbf{T Barry}. ``Collections in R: Review and Proposal.'' \textit{The R Journal 10.1}. \href{https://journal.r-project.org/archive/2018/RJ-2018-037/index.html}{Link}. & \textit{2018} \\ &
\end{tabular} 	
\end{rSection}


\begin{rSection}{TALKS}
	
	\begin{tabular}{p{14cm}R{2.5cm}}
		$\bullet$ ``Exponential family measurement error models for single-cell CRISPR screens.'' Joint Statistics meetings. Washington, DC. Contributed.  & \it{2021} \\
		
		$\bullet$  ``Conditional resampling improves calibration and sensitivity in single-cell CRISPR screen analysis.'' RECOMB-Seq. Held virtually. Contributed. & \it{2021} \\
		
		$\bullet$ ``Conditional resampling improves calibration and sensitivity in single-cell CRISPR screen analysis.'' Xin He Lab, University of Chicago. Held virtually. Invited. & \it{2021}
	\end{tabular}
	
\end{rSection}

\begin{rSection}{SERVICE AND SCIENCE OUTREACH}
	\begin{tabular}{p{14cm}R{2.5cm}}
		$\bullet$ Volunteer math, science, and English tutor to elementary and middle school students through UMD Lakeland STARS program. & \it{2015 -- 2018}  \\ 
		% $\bullet$ Volunteer mentor to a computer science undergraduate student (through CMU AI Undergraduate Research Mentoring program) & \it{2019}  \\ 
	\end{tabular} 
\end{rSection}

\begin{rSection}{COMPUTING}
	
	\begin{tabular}{p{14cm}R{2.5cm}}
		$\bullet$ Languages:  R, Python, C/C++, Nextflow  &   \\ 
		$\bullet$ Operating systems: Unix, Linux & \\
		$\bullet$ Clusters and clouds: SLURM, Sun Grid Engine, Azure & \\
		$\bullet$ Database systems: HDF5 & \\
		$\bullet$ Version control systems: Git/Github
	\end{tabular} 
	
	% \begin{tabular}{ @{} >{\bfseries}l @{\hspace{6ex}} l }
	%Computer Languages & Prolog, Haskell, AWK, Erlang, Scheme, ML \\
	%Protocols \& APIs & XML, JSON, SOAP, REST \\
	%Databases & MySQL, PostgreSQL, Microsoft SQL \\
	%Tools & SVN, Vim, Emacs
	% \end{tabular}
\end{rSection}

\textcolor{darkgray}{Updated January 2023}.

\end{document}