---
title: "CRISPR and single-cell CRISPR screens"
description: |
  CRISPR is a genome engineering tool that enables precise edits to the human genome. Single-cell CRISPR screens couple CRISPR to single-cell sequencing to power genome-scale causal inference.
author:
  - name: Tim Barry
    url: https://timothy-barry.github.io
output:
  distill::distill_article:
    self_contained: false
    toc: true
    toc_depth: 2  
categories:
  - Genomics
draft: false
preview: crispr_image.webpb
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


CRISPR is a genome engineering tool that has enabled scientists to edit and perturb human genomes with unprecedented precision and efficiency. Scientists are leveraging CRISPR to design medicines to ``fix'' aberrant genes in individuals suffering from genetic disorders; CRISPR-based therapeutics for sickle-cell anemia \citep{Frangoul2021}, cancer \citep{Yin2019}, and muscular dystrophy \citep{Min2019} are in development. Scientists also routinely apply CRISPR in non-clinical research settings to accelerate biological discovery. CRISPR screens \citep{Bock2022} are experiments in which CRISPR perturbations are introduced into a pool of cells, with each CRISPR perturbation targeting a region of the genome --- typically of unknown function --- for deactivation, inhibition, or activation. The effects of the induced CRISPR perturbations are assessed by monitoring the perturbed cells for phenotypic changes. For example, in the extreme case that all cells infected by a given perturbation die, one can infer that the region of the genome targeted by the given perturbation is crucial for cellular function. CRISPR screens have become a key component of drug discovery pipelines.

Single-cell sequencing is an experimental technique in which the molecular profile of a large number of individual cells is measured \citep{Hao2021}. First, a pool of cells is cultured; then, each cell is captured in a droplet containing chemical reagents, and the set of gene transcripts contained within each cell is sequenced. This process yields an $n \times p$ matrix, where $n$ is the number of cells sequenced and $p$ is the number of genes measured. (Typically, $n$ is on the order of several thousand to several million, while $p$ is on the order of $20,000$, the number of protein-coding genes in the human genome.) Each entry in the matrix is a count taking a value in the set of non-negative integers $\{0, 1, 2, \dots, \}.$ A given entry of the matrix indicates the expression level of a given gene within a given cell. Single-cell sequencing provides a more information-rich readout of the molecular content of a collection of cells than older methods, such as bulk RNA sequencing. Moreover, multimodal single-cell sequencing techniques enable the measurement of cellular modalities beyond gene expression, such as cell-surface protein expression and chromatin accessibility \citep{Zhu2020}.

Single-cell CRISPR screens combine CRISPR genome engineering and single-cell sequencing to survey the effects of CRISPR perturbations on individual cells \citep{Dixit2016,Mimitou2019,Papalexi2021}. In such screens, a library of CRISPR perturbations is introduced into a population of cells, following by single-cell sequencing to determine the perturbation (or perturbations) that each cell received and measure a rich molecular phenotype for each cell (typically including gene expressions). Single-cell CRISPR screens have emerged as among the most flexible and powerful methods for linking genetic perturbations to changes in cellular phenotypes, promising to accelerate cancer, longevity, and medical genetics research \citep{Replogle2022,Bock2022}. However, single-cell CRISPR screens pose considerable statistical challenges, currently limiting the reliability of conclusions drawn on the basis of single-cell CRISPR screen experiments. The primary objective of this thesis is to develop statistically rigorous and computationally efficient methods for the analysis of single-cell CRISPR screen data. The secondary objective of this thesis is to implement these methods in practical software tools aimed at working biologists. The methods that we develop --- which focus on robust hypothesis testing and estimation (with confidence) in the presence of measurement error --- may be of independent statistical interest.

## The mechanics of single-cell CRISPR screens

A single-cell CRISPR experiment involves several steps, which we describe here. First, the scientist constructs a library of CRISPR perturbations, each designed to target a different region of the genome for disruption. The CRISPR perturbations typically target genes \citep{Replogle2022} or enhancers \citep{Xie2017, Xie2019, Gasperini2019} with unknown or poorly understood function. (Enhancers are noncoding regulatory elements in the genome that are responsible for increasing or decreasing the expression of one or more nearby genes.) The CRISPR perturbations can knockout \citep{Dixit2016,Papalexi2021}, inhibit \citep{Schraivogel2020,Gasperini2019}, or activate \citep{Alda-Catalinas2020,Chardon2023} their targets; variants of CRISPR called CRISPR-knockout (CRISPRko), CRISPR-inhibition (CRISPRi), and CRISPR-activation (CRISPRa) implement these different modalities of genetic perturbation. Special perturbations called ``non-targeting'' (NT) perturbations are designed to exert no effect whatsoever on the cell and serve as negative controls. The CRISPR library typically consists of tens to tens of thousands of CRISPR perturbations.

Next, the scientist introduces the library of CRISPR of perturbations into a pool of cells; the perturbations assort into the cells randomly. The scientist typically chooses a cell type or tissue relevant to the disease or trait under investigation. (For example, if the scientist is interrogating the immune system, the scientist likely would select immune cells, such as T cells, for perturbation; \citealt{Frangieh2021}.) The perturbations can be administered either at ``high multiplicity-of-infection'' (high-MOI; \citealt{Gasperini2019,Morris2023, Yao2023}) or ``low multiplicity-of-infection'' (low-MOI; \citealt{Dixit2016,Replogle2022}). In high-MOI screens many (e.g., 10-30) CRISPR perturbations are inserted into each cell; in low-MOI screens, by contrast, exactly one CRISPR perturbation is inserted into each cell. (Cells that by chance receive two or more perturbations are discarded in low-MOI screens.) The high-MOI design is more common for enhancer-targeting screens (as each perturbation is expected to have a weak effect in that setting), while the low-MOI design is more common for gene-targeting screens.

The scientist then waits several days for the perturbations to take effect and uses single-cell sequencing to measure each cell's transcriptome (i.e., its gene expressions) and the perturbation (or perturbations) that it received. (Each cell's protein expressions and/or chromatin accessibility profile may also be measured; \citealt{Frangieh2021,Papalexi2021,Liscovitch2021}.) Finally, the scientist subjects the data to a statistical analysis. The objective of the statistical analysis is to understand how the various perturbations impact the expressions of the genes. There are two broad approaches to analyzing single-cell CRISPR screen data: multivariate and univariate. The multivariate approach \citep{Zhou2022,Yao2023} involves fitting a model to the entire set of perturbations and gene expressions and leveraging this fitted model to study the impact of perturbations (or sets of perturbations) on genes (or sets of genes). The univariate approach \citep{Gasperini2019,Barry2021}, by contrast, involves testing for association (and estimating the change in expression) for a large set of preselected perturbation-gene pairs. (Each pair is analyzed one at a time). The multivariate and univariate approaches each present statistical and computational advantages and disadvantages. This thesis focuses on the univariate approach, which is simpler (though still quite challenging), more commonplace, and more amenable to computational parallelization.

## The univariate approach to analyzing single-cell CRISPR screen data

We describe the univariate approach to single-cell CRISPR screen analysis in greater detail. The analysis procedure differs somewhat between low-MOI and high-MOI settings. To test for association between a given targeting CRISPR perturbation and gene in low-MOI (Figure \ref{fig:intro_cartoon}a), one first divides the cells into two groups: those that received the targeting perturbation, and those that received a non-targeting (NT) perturbation. (All other cells  typically are ignored.) One then tests for differential expression of the given gene across these two groups of cells, yielding a fold change estimate and $p$-value. (The differential expression test entails testing if the distribution of the gene expression differs across the two groups of cells.) One repeats this procedure for a (typically) large, preselected set of perturbation-gene pairs. Finally, one computes the discovery set by subjecting the tested pairs to a multiplicity correction procedure, typically the BH procedure or a variant thereof \citep{BH95,BY01}. The procedure for analyzing high-MOI single-cell CRISPR screen data is similar. To test for association between a given targeting perturbation and gene, one partitions the cells into two groups: those that received the given targeting perturbation, and those that did \textit{not} receive the given targeting perturbation. One then tests for differential expression of the given gene across these two groups of cells (Figure \ref{fig:intro_cartoon}b). (One cannot compare against cells containing only NT perturbations, as very few (if any) cells contain only NT perturbations in high-MOI.)

\begin{figure}[h]
\includegraphics[width = \textwidth]{chapters/chapter-0-figs/ch0_fig.png}
\caption{\textbf{Overview of the statistical analysis protocol for single-cell CRISPR screen data}. \textbf{a}, The standard paradigm for association testing on low-MOI single-cell CRISPR screen data. One compares the expression of a given gene across two groups of cells: those containing the given targeting perturbation, and those containing a non-targeting (NT) perturbation. \textbf{b}, The standard paradigm for high-MOI data. One compares the expression of the given gene across cells containing the given targeting perturbation and \textit{all other cells}. \textbf{c}, The relationship between the variables in the analysis. The technical factors may or may not exert a confounding effect on the perturbation indicator and gene expression.}
\label{fig:intro_cartoon}
\end{figure}

## A first-pass statistical framing of the problem

We provide a first-pass, high-level statistical rendering of the problem to make the problem more concrete for a statistical audience. Consider a given targeting-perturbation-gene pair. We observe i.i.d.\ data $(X_1, Y_1, Z_1), \dots, (X_n, Y_n, Z_n)$, where $X_i \in \{0,1\}$ is the ``perturbation indicator'' of cell $i$, $Y_i \in \{0, 1, 2, 3, \dots \}$ is the expression of the gene in cell $i$, and $Z_i \in \R^p$ is a low-dimensional vector of ``technical factors'' measured in cell $i$. $X_i$ is set to $1$ if cell $i$ contains the targeting perturbation and $0$ otherwise. (In low-MOI, $X_i = 0$ if cell $i$ contains a non-targeting perturbation, while in high-MOI, $X_i = 1$ if cell $i$ does \textit{not} contain the targeting perturbation.) The ``technical factor'' vector $Z_i$ is a vector of cell-specific covariates that is measured as part of the single-cell sequencing process. The technical factor vector includes covariates such as sequencing depth (i.e., the number of gene transcripts sequenced in the cell) and sequencing batch (i.e., the batch in which the given cell was sequenced). Typically, the dimension of $Z_i$ ranges from about three to seven. We assume that the technical factor vector $Z_i$ \textit{may} or \textit{may not} exert a confounding effect on the perturbation indicator $X_i$ and gene expression count $Y_i$. (That is, $X_i$ and $Y_i$ may be dependent or independent of $Z_i$) Whether or not $Z_i$ exerts a confounding effect can vary across datasets and even across pairs within a given dataset.

Rendered in these terms, our primary objective is to test whether the gene expression count is independent of the perturbation indicator given the technical factor vector (Figure \ref{fig:intro_cartoon}c). In other words, we seek to test the conditional independence null hypothesis $H_0: Y_i \indep X_i | Z_i.$ (Note that if $Z_i$ does not exert a confounding effect, then conditional independence $Y_i \indep X_i | Z_i$ and marginal independence $Y_i \indep X_i$ hypotheses are equivalent.) We seek to develop a method for this purpose that is \textit{well-calibrated} and \textit{powerful}. That is, the method should produce uniformly distributed p-values under the null hypothesis of no association between the perturbation indicator and the expression of the gene, and the method should yield small p-values when there \textit{is} an association.

Two methods frequently employed to carry out the association test are negative binomial regression \citep{Gasperini2019} and the Wilcoxon two-sample test \citep{Mimitou2019,Papalexi2021,Wessels2022}. Negative binomial regression models the gene expressions as a function of the perturbation indicator and technical factors:
\begin{equation}\label{eqn:into_nb_model}
Y_i \sim \textrm{NB}_\theta(\mu_i); \quad \log(\mu_i) = \gamma X_i + \beta^T Z_i,
\end{equation}
where $NB_\theta(\mu_i)$ denotes the negative binomial distribution with size parameter $\theta$ and mean $\mu_i$, and $\gamma \in \R$ and $\beta^T \in \R^p$ are unknown constants. The model (\ref{eqn:into_nb_model}) is fitted to the data, and a Wald test is used to test the hypothesis $H_0: \gamma = 0.$ (If the NB model is correctly specified, then a test of the null hypothesis $H_0: \gamma = 0$ is equivalent to a test of the conditional independence hypothesis $H_0: X_i \indep Y_i | Z_i$.) The Wilcoxon two-sample test, meanwhile, nonparametrically tests for a distributional difference across the sets $\{Y_i : X_i = 1\}$ and $\{Y_i : X_i = 0\}$. (The vector $Z_i$ is ignored.) While negative binomial regression and the Wilcoxon test are decent starting points, both can be improved upon considerably via more sophisticated techniques, as we aim to show.

Our secondary goal is to estimate the effect of the perturbation indicator $X_i$ on the gene expression count $Y_i$ while controlling for the technical factor vector $Z_i$. This secondary goal moves beyond statistical significance and toward biological significance. Within the context of the negative binomial model (\ref{eqn:into_nb_model}), the estimation problem amounts to estimating (with confidence) the parameter $\gamma$. Chapters \ref{ch:one} and \ref{ch:three} focus on the hypothesis testing problem, while Chapter \ref{ch:two} focuses on the estimation problem.

## A key application of single-cell CRISPR screens: illuminating the function of noncoding GWAS variants

Single-cell CRISPR screens are poised to impact a variety of research areas. One especially important application of single-cell CRISPR screens is to shed light on results obtained from genome-wide association studies (GWAS). The purpose of a GWAS is to identify regions in the genome that are associated with a trait or disease of interest. A GWAS proceeds in several steps \citep{Uffelmann2021}. First, one identifies a disease of interest (e.g., bipolar disorder). Next, one collects a large sample of subjects (typically numbering in the hundreds of thousands), where some of these subjects are cases (i.e., they have the disease) and the rest are controls (i.e., they are healthy). One then genotypes each individual, which consists of sequencing each individual's DNA at a large number of prespecified locations within the genome (called \textit{variants}). This genotyping step yields an $n \times p$ matrix, where $n$ is the number of subjects in the study, and $p$ is the number of variants sequenced. Each entry of the matrix --- coded as 0, 1, or 2 --- indicates the version of each variant that each individual harbors. Finally, one regresses the disease status vector onto the matrix of variants, typically proceeding one variant at a time. This process yields a $p$-value for the test of association between each variant and the disease. Variants with very small $p$-values are considered to be associated with the disease. One typically plots the $p$-value of each variant against its genomic position, forming a so-called ``Manhattan plot.'' A Manhattan plot for bipolar disorder is shown in Figure \ref{fig:gwas_example}.

\begin{figure}[h]
\centering
\includegraphics[width = 0.8\textwidth]{chapters/chapter-0-figs/gwas_example.png}
\caption{\textbf{An example GWAS of bipolar disorder}. Each point represents a genomic variant. The $x$-axis displays the position of the genomic variant in the genome, and the $y$-axis displays the $p$-value for a test of association between the genomic variant and the phenotype (i.e., presence of absence of bipolar disorder). The horizontal red line indicates the genome-wide significance level, drawn at $5 \cdot 10^{-8}$. Variants with $p$-values below the genome-wide significance threshold are considered to be significantly associated with the phenotype. (Figure taken from \citealt{Mullins2021}).}
\label{fig:gwas_example}
\end{figure}

Two major challenges associated with interpreting GWAS results are as follows \citep{Uffelmann2021,Tam2019}. First, the vast majority (>$98\%$) of variants associated with diseases are located outside of genes and in noncoding regions \citep{Gallagher2018}. These noncoding variants are thought to lie inside of enhancers, contributing to disease by modulating the expression of one or more disease-relevant genes. It is unclear which gene (or genes) most noncoding variants regulate. A major objective in contemporary genetics, therefore, is to map noncoding variants to target genes at genome-wide scale. Second, variants in close physical proximity tend to be extremely highly correlated with one another (due to linkage disequilibrium), making it challenging to identify the causal variant among a set of nearby variants. A second major objective in contemporary genetics, therefore, is to distinguish causal variants from variants that merely are strongly correlated with causal variants.

Single-cell CRISPR screens are among the most promising technologies for resolving both of these challenges \citep{Morris2023}. Several recent, cutting edge studies have leveraged single-cell CRISPR screens to map noncoding GWAS variants \citep{Morris2023,Tuano2023}. These studies targeted CRISPR perturbations to noncoding GWAS variants in disease-relevant cell types. These GWAS variants were then mapped to target genes by testing for association between the CRISPR perturbations that disrupted these GWAS variants and nearby genes. The former study used CRISPR base editing to disrupt single variants among sets of variants in strong linkage disequilibrium and mapped these individual variants. Altogether, these studies demonstrate the promise of single-cell CRISPR screens to elucidate the function of noncoding GWAS variants.
